src/main/java/com/google/firebase/database/connection/WebsocketConnection.java

```diff
@@ -324,8 +324,9 @@ public void run() {
 
     @Override
     public void onError(final Throwable e) {
-      if (e.getCause() != null && e.getCause() instanceof EOFException) {
-        logger.error("WebSocket reached EOF", e);
+      if (e instanceof EOFException
+          || (e.getCause() != null && e.getCause() instanceof EOFException)) {
```

Comment:
You don't need to perform a null check before calling instanceof.

Reply:
Done